ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -215,6 +257,12 @@ def _reset_env(self, env: BaseUnityEnvironment):
                 train_mode=self.fast_simulation,
                 config=self.meta_curriculum.get_config(),
             )
+        elif self.sampler_manager is not None:
+            new_config = self.sampler_manager.sample_all()
```

Comment:
If you always create the SamplerManager, you need the `else` branch. Passing an empty dict for the config should be equivalent to not passing one.

I think you can simplify this whole function to something like
```python
new_config = self.sampler_manager.sample_all()
meta_curriculum_config = self.meta_curriculum.get_config() if self.meta_curriculum else {}
new_config.update(meta_curriculum_config)

return env.reset(train_mode=self.fast_simulation, config=new_config)
```